### ISSUE TITLE
weird visual glitch causing low fps

### ISSUE THREAD

**Issue by EliteDiego** [bluemagic123/tModLoader#4398]
### Version

Latest 1.4.4-stable

### OS

Windows

### Platform

GOG

### This bug affects...

Gameplay as a Player

### Description

had to record on my phone because i cant on my laptop

https://github.com/user-attachments/assets/c60d264c-8cbe-4ec6-92cf-3e58483be0cf

update: so as you can see on the video, i got this "visual" glitch. and honestly, i dont know what to do. this also happens on normal terraria when im on fullscreen mode with trippy/retro, also lows my fps to 30. plz help

mods im using:

  "ShimmerExpansion"
  "endless"
  "FargoSeeds"
  "LootBonus"
  "ImprovedReforging"
  "SummonWhips"
  "ImproveGame"
  "miningcracks_take_on_luiafk"
  "MoreAccessories"
  "OreExcavator"
  "RecipeBrowser"
  "SerousCommonLib"
  "BossChecklist"
  "AlchemistNPCLite"
  "MagicStorage"
  "MagicBuilder"
  "imkSushisMod"
  "TerraWeapons"
  "WMITF"
  "Dodgeroll"
  "ExtraLoadouts"
  "Aimbot"
  "PrimeRework"
  "PointShop"
  "StructureHelper"
  "MeleeEffects"
  "WeaponDisplayLite"
  "ConciseModList"
  "CheatSheet"
  "CheatExtensions"
  "HEROsMod"
  "LootBeams"
  "TerrariaAmbienceAPI"
  "TerrariaAmbience"
  "BetterGameUI

### Log File
[client.log](https://github.com/user-attachments/files/17267969/client.log)
[environment-client.log](https://github.com/user-attachments/files/17267970/environment-client.log)
[Launch.log](https://github.com/user-attachments/files/17267971/Launch.log)
[Natives.log](https://github.com/user-attachments/files/17267972/Natives.log)
[terrariasteamclient.log](https://github.com/user-attachments/files/17267973/terrariasteamclient.log)
### Steps to reproduce
1. put yourself on fullscreen mode
2. put lighting mode on trippy or retro
3. put zoom on 100% an do what i do on the video
4. see what happens

### Expected Behavior
maybe my laptop causing this

### Actual Behavior

put my fps to a cap of 30

### Reproduction frequency

ALL the time. i think theres no fix....
### Additional Information

basically ALL of this happens on fullsrceen. every time i put color lighting i get to a cap of 30fps when always using color/white. in the case of using trippy/retro when you are on minimum zoom i get 60fps BUT when you zoom IN a little bit everything dissapears for a second and my fps gets capped on 30fps, BUT when im on window mode i always get 60 but still happens the thing that dissapears everything for a second (lighting mode on trippy/retro)

(sorry if someone dont understand this but my english is so bad)

**Comment by JavidPack** [comment]
There is a lot going on here. 

I'm seeing errors from CombinationsMod and SummonWhips in your log, but I'm not positive how they are affecting things. 

The log you posted has mods and resource packs enabled, so I can't really use it to judge if something is wrong with tmodloader itself. You mentioned having low FPS issues at minimum zoom without any mod or resource pack enabled, for this I can only assume that your GPU, "Intel(R) UHD Graphics", isn't that powerful and is struggling. You can read this for some things to try: https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#low-fps

You mentioned disappearing tiles, that usually means an exception which will show up in the log, but it's not in the log you posted. That is probably some mod with bad logic, you'll need to look at client.log after experiencing the issue again to find it.

You have BetterZoom enabled, so any buggy zoom situation could be caused by that.

You can always delete the config file (C:\Users\Alumno\Documents\My Games\Terraria\tModLoader\config.json) as well to revert to default settings, if you think a setting is causing an issue.

**Comment by EliteDiego** [comment]
> There is a lot going on here.
> 
> I'm seeing errors from CombinationsMod and SummonWhips in your log, but I'm not positive how they are affecting things.
> The log you posted has mods and resource packs enabled, so I can't really use it to judge if something is wrong with tmodloader itself. You mentioned having low FPS issues at minimum zoom without any mod or resource pack enabled, for this I can only assume that your GPU, "Intel(R) UHD Graphics", isn't that powerful and is struggling. You can read this for some things to try: https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#low-fps
> 
> You mentioned disappearing tiles, that usually means an exception which will show up in the log, but it's not in the log you posted. That is probably some mod with bad logic, you'll need to look at client.log after experiencing the issue again to find it.
> 
> You have BetterZoom enabled, so any buggy zoom situation could be caused by that.
> 
> You can always delete the config file (C:\Users\Alumno\Documents\My Games\Terraria\tModLoader\config.json) as well to revert to default settings, if you think a setting is causing an issue.



> There is a lot going on here.
> 
> I'm seeing errors from CombinationsMod and SummonWhips in your log, but I'm not positive how they are affecting things.
> 
> The log you posted has mods and resource packs enabled, so I can't really use it to judge if something is wrong with tmodloader itself. You mentioned having low FPS issues at minimum zoom without any mod or resource pack enabled, for this I can only assume that your GPU, "Intel(R) UHD Graphics", isn't that powerful and is struggling. You can read this for some things to try: https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#low-fps
> 
> You mentioned disappearing tiles, that usually means an exception which will show up in the log, but it's not in the log you posted. That is probably some mod with bad logic, you'll need to look at client.log after experiencing the issue again to find it.
> 
> You have BetterZoom enabled, so any buggy zoom situation could be caused by that.
> 
> You can always delete the config file (C:\Users\Alumno\Documents\My Games\Terraria\tModLoader\config.json) as well to revert to default settings, if you think a setting is causing an issue.


alright, i did everything you say. i did an update on the post, but still have a problem

**Comment by JavidPack** [comment]
I tested Terraria and the flicker when zooming in from 100% in trippy/retro also happens there. It seems to be unavoidable with the design of the internal rendering pipeline from what I could interpret. The flicker seems to be a vanilla Terraria issue, and not one that is easily fixed by what we can do in tModLoader.


https://github.com/user-attachments/assets/1525895a-a64a-4ad8-97ba-e5fbc8ce36b7


This may just be unavoidable, luckily zooming in during gameplay isn't typical, so it shouldn't affect things very much.

Your frame rate issue, however, seems to just be a limitation of your CPU/GPU capabilities. I see that your computer has 2 cores, 4 GB RAM, and integrated GPU. Given the low specs and the additional overhead needed for tModLoader and mods, I'm willing to bet consistent 60 fps might be difficult for your computer. Even if normal Terraria is 60fps in the same situations, tModLoader needs a bit more due to all the additional work that is being done.

**Comment by EliteDiego** [comment]
> I tested Terraria and the flicker when zooming in from 100% in trippy/retro also happens there. It seems to be unavoidable with the design of the internal rendering pipeline from what I could interpret. The flicker seems to be a vanilla Terraria issue, and not one that is easily fixed by what we can do in tModLoader.
> 
>  Terraria_2024-10-07_15-21-29.mp4 
> This may just be unavoidable, luckily zooming in during gameplay isn't typical, so it shouldn't affect things very much.
> 
> Your frame rate issue, however, seems to just be a limitation of your CPU/GPU capabilities. I see that your computer has 2 cores, 4 GB RAM, and integrated GPU. Given the low specs and the additional overhead needed for tModLoader and mods, I'm willing to bet consistent 60 fps might be difficult for your computer. Even if normal Terraria is 60fps in the same situations, tModLoader needs a bit more due to all the additional work that is being done.

welp... i think the solution is to get a better laptop.  but i thing was causing to cap my fps was an external v-sync from my laptop but it seems that my fps are now at 50-55 anyway, thanks for the help. but theres a way to optimize color lighting? the reason why was using trippy was for the low fps i had with color light, or maybe something that can make me get more zoom without having to use mods?

**Comment by JavidPack** [comment]
I'm not sure if by more zoom you mean more zoom out or more zoom in, but in My Games/Terraria/tModLoader/config.json there is a RemoveForcedMinimumZoom option you can change to true, that should let you zoom out more, I think.

There is a mod called Nitrate that claims to improve frame rate, you can try it.

**Comment by JavidPack** [comment]
This issue has either no recent activity or no reproducible steps, so it is being closed. We assume this issue has solved itself in some manner. Feel free to open a new issue if the issue persists and fill out the issue template completely.
